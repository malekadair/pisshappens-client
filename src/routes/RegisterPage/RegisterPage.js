import React, { Component } from 'react'
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'
import Nav from '../../components/Nav/Nav';
import Footer from '../../components/Footer/Footer';

class RegisterPage extends Component {
	static defaultProps = {
		history: {
			push: () => { },
		},
	}

	handleRegistrationSuccess = () => {
		const { history } = this.props;
		history.push('/login');
	}

	render() {
		return (
			<section className='RegistrationPage'>
				<header>
					<Nav />
				</header>
				<RegistrationForm
					onRegistrationSuccess={this.handleRegistrationSuccess} />
				<Footer />
			</section>
		)
	}
}

export default RegisterPage