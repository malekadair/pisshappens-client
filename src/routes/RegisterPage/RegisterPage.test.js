import React from "react";
import toJson from "enzyme-to-json";
import RegisterPage from "./RegisterPage";

describe(`RegisterPage component`, () => {
	const props = {
		id: "a",
		name: "test-class-name"
	};

	it("renders a RegisterPage by default", () => {
		const wrapper = <RegisterPage />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("renders the RegisterPage given props", () => {
		const wrapper = <RegisterPage {...props} />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});