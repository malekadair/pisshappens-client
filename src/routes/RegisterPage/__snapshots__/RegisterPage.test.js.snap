// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage component renders a RegisterPage by default 1`] = `undefined`;

exports[`RegisterPage component renders the RegisterPage given props 1`] = `undefined`;
